addons/evm/src/signers/secret_key.rs

```diff
@@ -325,6 +327,7 @@ impl SignerImplementation for EvmSecretKeySigner {
                 )
             })?;
             let tx_nonce = tx_envelope.nonce();
+            let tx_chain_id = tx_envelope.chain_id().unwrap_or(u64::MAX);
```

Comment:
I'm not sure I understand the rationale behind this `u64::MAX` fall back.

Reply:
oh I see.

Reply:
Very true, fixed! An `unwrap` is safe here - we always require the user provide a chain id, and it is always set on the transactions we build.